src/D2L.Bmx/BmxConfigProvider.cs

```diff
@@ -75,23 +74,30 @@ internal class BmxConfigProvider : IBmxConfigProvider {
 
 	// Look from cwd up the directory chain all the way to root for a .bmx file
 	private static FileInfo? GetProjectConfigFileInfo() {
-		DirectoryInfo? currentDirectory = new( Directory.GetCurrentDirectory() );
+		try {
+			DirectoryInfo? currentDirectory = new( Directory.GetCurrentDirectory() );
 
-		while( currentDirectory is not null ) {
-			var bmxConfigFiles = currentDirectory.GetFiles( ".bmx" );
+			// limit the maximum search depth to avoid infinite loop even if there's any symlink trickery
+			for( int i = 0; i < 2000 && currentDirectory is not null; i++ ) {
+				var bmxConfigFiles = currentDirectory.GetFiles( ".bmx" );
 
-			if( bmxConfigFiles.Length > 1 ) {
-				// TODO: Log this
-				return null;
-			}
+				if( bmxConfigFiles.Length > 1 ) {
+					// TODO: Log this
```

Comment:
Don't think we will do this?

Reply:
Maybe?
I can conceive a scenario where a user might complain "why isn't .bmx config working for me", and we'll instruct them to enable some debug flag (like [this](https://github.com/Brightspace/bmx/pull/325/files#diff-707f49706d83b60f0cc24120487272507787e1c1144b484d1bdfb1d43eb21205R168-R169)) that'll turn on optional debug logging here and we can help them resolve issues more easily.